src/core/sagas/languageTools.js

```diff
@@ -0,0 +1,40 @@
+// Disabled because of
+// https://github.com/benmosher/eslint-plugin-import/issues/793
+/* eslint-disable import/order */
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+/* eslint-enable import/order */
+
+import { languageTools as languageToolsApi } from 'core/api/languageTools';
+import log from 'core/logger';
+import {
+  FETCH_LANGUAGE_TOOLS,
+  loadLanguageTools,
+} from 'core/reducers/addons';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+
+export function* fetchLanguageTools({
+  payload: { errorHandlerId },
+}) {
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const state = yield select(getState);
+
+    const response = yield call(languageToolsApi, {
+      api: state.api,
+      auth: true,
+    });
+
+    yield put(loadLanguageTools({ addons: response.results }));
```

Comment:
Since these are just add-ons (with a reduced fieldset) you should call `loadAddons()` here. I don't think a separate `loadLanguageTools()` action is necessary. This will cause the add-on objects to always be stored with `createInternalAddon()` which I think is a good idea. We will then have a consistent representation of all add-on objects even if a bunch of the fields are null or undefined. This also means you would need to update `createInternalAddon()` so that it sets the language tool specific fields to null or undefined when they aren't set.

Reply:
Using loadAddons means I have to use normalizr or set up the entities weirdness to get the actions to work. I really didn't want to do that as we want to get closer to using the responses from the API.

I would rather add a TODO: that says this can be converted to loadAddons once #2917 is closed.

Reply:
ok, then I would suggest making a new action creator called `loadAddonsFromResults()` with a TODO comment in `loadAddons()` saying that it is deprecated. The key change I'd like to see is that the reducer uses `createInternalAddon(addons[key])` not `createInternalLanguageTool(addons[key])` (I don't think we need that function).

Reply:
The next step would be this:

> ...you would need to update createInternalAddon() so that it sets the language tool specific fields to null or undefined when they aren't set.

This enables every function working with the state data to always gets the same object, no matter what type of add-on it is. This will cut down on surprises.